cmd/podman/system_prune.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/containers/libpod/cmd/podman/shared"
+	"github.com/containers/libpod/libpod/adapter"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/urfave/cli"
+)
+
+var (
+	pruneSystemDescription = `
+	podman system prune
+
+        Remove unused data
+`
+	pruneSystemFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "all, a",
+			Usage: "remove all unused data",
+		},
+		cli.BoolFlag{
+			Name:  "force, f",
+			Usage: "Do not prompt for confirmation",
+		},
+		cli.BoolFlag{
+			Name:  "volumes",
+			Usage: "Prune volumes",
+		},
+	}
+	pruneSystemCommand = cli.Command{
+		Name:         "prune",
+		Usage:        "Remove unused data",
+		Description:  pruneSystemDescription,
+		Action:       pruneSystemCmd,
+		OnUsageError: usageErrorHandler,
+		Flags:        pruneSystemFlags,
+	}
+)
+
+func pruneSystemCmd(c *cli.Context) error {
+
+	// Prompt for confirmation if --force is not set
+	if !c.Bool("force") {
+		reader := bufio.NewReader(os.Stdin)
+		volumeString := ""
+		if c.Bool("volumes") {
+			volumeString = `
+        - all volumes not used by at least one container`
+		}
+		fmt.Printf(`
+WARNING! This will remove:
+        - all stopped containers%s
+        - all dangling images
+        - all build cache
+Are you sure you want to continue? [y/N] `, volumeString)
+		ans, err := reader.ReadString('\n')
+		if err != nil {
+			return errors.Wrapf(err, "error reading input")
+		}
+		if strings.ToLower(ans)[0] != 'y' {
+			return nil
+		}
+	}
+
+	runtime, err := adapter.GetRuntime(c)
+	if err != nil {
+		return errors.Wrapf(err, "could not get runtime")
+	}
+	defer runtime.Shutdown(false)
+
+	ctx := getContext()
+	fmt.Println("Deleted Containers")
+	err = pruneContainers(runtime, ctx, shared.Parallelize("rm"), c.Bool("force"))
+	if c.Bool("volumes") {
+		fmt.Println("Deleted Volumes")
+		err1 := volumePrune(runtime, getContext())
+		if err1 != nil {
+			if err != nil {
+				logrus.Errorf("%q", err1)
+			} else {
+				err = err1
+			}
+		}
+	}
+
+	// Call prune; if any cids are returned, print them and then
+	// return err in case an error also came up
+	pruneCids, err1 := runtime.PruneImages(c.Bool("all"))
```

Comment:
s/err1/err/PruneImages/ ... just to make the code easier to brain-parse

Reply:
Ok I changed to use lasterr to make it clearer.